src/app/agent/agent-dashboard/agent-dashboard.component.ts

```diff
@@ -27,31 +30,123 @@ export class AgentDashboardComponent implements OnInit {
   ) {}
 
   ngOnInit() {
+    document.getElementById("nav-agent").classList.add("active");
     this.username = localStorage.getItem("username");
+    this.airportsMap = new Map();
 
-    this.loadAgent();
-    this.loadFlights();
     this.loadAirports();
+    this.loadFlights();
+    this.loadAgent();
+    
+    this.bookFlightPage = true;
+    this.cancelFlightPage = false;
+  }
 
-    this.traveler = {
-      active: true,
-      bookerId: 1,
-      travelerId: 2,
-    };
+  ngOnDestroy() {
+    document.getElementById("nav-agent").classList.remove("active");
   }
 
   loadAgent() {
     this.service
-      .get("http://127.0.0.1:8083/agent/users/" + this.username)
+      .get(
+        `${environment.agentBackendUrl}${environment.usernameUri}/${this.username}`
+      )
+      .subscribe((result) => {
+        this.agent = result;
+
+        this.service
+          .get(
+            `${environment.agentBackendUrl}${environment.bookingsUri}/${this.agent.userId}`
+          )
+          .subscribe(
+            (result) => {
+              this.bookings = result;
+
+              this.bookings.forEach((element) => {
+                element = this.processBooking(element);
+              });
+            },
+            (error) => {
+              alert(error);
+            }
+          );
+      });
+  }
+
+  processBooking(booking) {
+    booking.name = this.setTravelerName(booking);
+    booking.flight = this.setFlight(booking);
+  }
+
+  setTravelerName(input: any) {
+    this.service
+      .get(
+        `${environment.agentBackendUrl}${environment.userIdUri}/${input.travelerId}`
+      )
+      .subscribe(
+        (result: any) => {
+          input.name = result.name;
+        },
+        (error: any) => {
+          input.name = "(error retrieving traveler name)";
+          // do something with logger here
+        }
+      );
+  }
+
+  setAirports(input: any) {
+    input.arriveAirport = this.airportsMap.get(input.flight.arriveId);
+    input.departAirport = this.airportsMap.get(input.flight.departId);
+  }
+
+  setFlight(input: any) {
+    this.service
+      .get(
+        `${environment.agentBackendUrl}${environment.flightUri}/${input.flightId}`
+      )
+      .subscribe(
+        (result: any) => {
+          input.flight = result;
+
+          this.setAirports(input);
+          input.flight.departTime = moment(input.flight.departTime).format(
+            "MMMM Do YYYY, h:mm:ss a"
+          )
+        },
+        (error: any) => {
+          input.flight = "(error retrieving arrive airport name)";
```

Comment:
This will result in blanks (undefined) when the template tries to display b.flight.departTime.